src/sknnr/_gnn.py

```diff
@@ -1,12 +1,94 @@
+from __future__ import annotations
+
+from typing import Callable, Literal
+
 from sklearn.base import TransformerMixin
+from sklearn.metrics import DistanceMetric
 
 from ._base import TransformedKNeighborsRegressor, YFitMixin
 from .transformers import CCATransformer
 
 
 class GNNRegressor(YFitMixin, TransformedKNeighborsRegressor):
-    def __init__(self, n_components=None, **kwargs):
-        super().__init__(**kwargs)
+    """
+    Regression using Gradient Nearest Neighbor (GNN) imputation.
+
+    The target is predicted by local interpolation of the targets associated with
+    the nearest neighbors in the training set, calculated in transformed Canonical
+    Correspondence Analysis (CCA) space.
+
+    See `sklearn.neighbors.KNeighborsRegressor` for more information on parameters
+    and implementation.
+
+    Parameters
+    ----------
+    n_neighbors : int, default=5
+        Number of neighbors to use by default for `kneighbors` queries.
+    n_components : int, default=None
```

Comment:
Great question that I wasn't sure on either. [`numpydoc`](https://numpydoc.readthedocs.io/en/latest/format.html#parameters) doesn't seem to include `None` as a parameter type in these cases, even though it would be needed in the type annotation itself. That probably makes sense from a readability perspective, since `default=None` implies that `None` is supported.  

While I was looking into that, I noticed that `numpydoc` *does* distinguish between `default=None` and `optional`, and suggests the latter when `None` signifies an undefined parameter, e.g.:

```
n_components : int, optional
```

When `None` is directly used in the function, they suggest specifying `default=None`, e.g.:

```
metric_params : dict, default=None
```

Based on that, it looks like `n_components` and `n_jobs` should definitely be marked `optional`, whereas `metric_params` probably makes sense as `default=None`. I'll make those changes.

Reply:
That's great.  Thanks for digging into `numpydoc`.